### PR TITLE
Subgroup-specific cross-validation

### PR DESCRIPTION
**Pull request by Lauren-EylerDang**
Hello,
I added a function folds_subgroup which allows for subgroup-specific cross-validation. In this case, the validation set is comprised entirely of the subgroup of interest, and the training set consists of the rest of the subgroup of interest and all other subgroups' data. I am using this function for a project where the goal is to train an optimal model for a subgroup while using data from other groups. This is my first pull request, so I apologize if there is anything I did not do correctly! I tried to follow the instructions and am happy to fix anything (eg formatting) as needed. Thank you!
-Lauren

### CODE REVIEW

**Comment by tdhock** [comment]
Hi Lauren, your code seems related to the SOAK algorithm we recently published, https://onlinelibrary.wiley.com/doi/10.1002/sam.70055
and code is available in mlr3resampling, https://mlr3resampling.netlify.app/doc/soaked
